src/Database/Attach/File.php

```diff
@@ -271,10 +257,8 @@ public function setDataAttribute($value)
      * Helper attribute for get image width.
      *
      * Returns `null` if this file is not an image.
-     *
-     * @return string|int|null
      */
-    public function getWidthAttribute()
+    public function getWidthAttribute(): string|int|null
```

Comment:
Could you consider changing the width and height attribute to be optional when using an s3 driver in some form. These 2 attributes make any remote storage take more than 7 seconds because it has to download the file locally.

Reply:
actually nvm, I can just override it

Reply:
We could store that data on upload theoretically and retrieve it from metadata instead of recalculating

Reply:
hmm, if that's the case, wouldn't it also be beneficial to have a `has_thumb` meta parameter to indicate if the image has a thumbnail or not.

Currently the way I have it is that I have a `has_thumb` property on the file model. so rather than using `hasFile` which checks the remote file location it checks if that parameter is set, then I offload the thumbnail generation to a queue (this is more of a custom solution for me that im utilizing, but using `hasFile` slows down the process a lot.)

Reply:
We should probably add a `metadata` column in the form of a JSON blob to store this sort of information